Task-WebRTC.md

```diff
@@ -0,0 +1,370 @@
+security layer for data channels that are available to users. The 
+signalling channel will persist after being handed over to a dedicated 
+data channel once the peer-to-peer connection has been set up. 
+Therefore, further signalling communication between the peers does not 
+require a dedicated WebSocket connection over a SaltyRTC server.
+
+# Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC 2119 [RFC2119]](https://tools.ietf.org/html/rfc2119).
+
+# Terminology
+
+All terms from the 
+[SaltyRTC protocol specification](./Protocol.md#terminology) are valid 
+in this document. Furthermore, this document will reference API parts 
+of the [WebRTC specification](https://www.w3.org/TR/webrtc/).
+
+## Wrapped Data Channel
+
+This protocol adds another security layer for WebRTC data channels as 
+we want our protocol to sustain broken (D)TLS environments. A data 
+channel that uses this security layer will be called *Wrapped Data 
+Channel*. Note that the user MAY or MAY NOT choose to use wrapped data 
+channels for its purpose. However, the signalling channel MUST be 
+handed over to a wrapped data channel.
+
+# Task Protocol Name
+
+The following protocol name SHALL be used for task negotiation:
+
+`v0.webrtc.tasks.saltyrtc.org`
+
+TODO: Switch to `v1` as soon as the spec has been reviewed.
+
+# Detecting the Maximum Message Size
+
+For now, all implementations SHALL use the value `16384` which seems 
+to be the highest amount of kilobytes that can be applied for portable 
+WebRTC data channel communication. An implementation MAY use another 
+value if it can guarantee delivery and reception for messages of that 
+size. A value of `0` indicates that the implementation is able to 
+handle messages of arbitrary length (hooray!).
+
+# Task Data
+
+This task makes use of the *data* field of the 'auth' messages 
+described in the 
+[SaltyRTC protocol specification](./Protocol.md#auth-message).  
+The *Outgoing* section describes what the data of this task SHALL 
+contain and how it MUST be generated. Whereas the *Incoming* section 
+describes how the task's data from the other client's 'auth' message 
+MUST be validated and potentially stored.
+
+## Outgoing
+
+The task's data SHALL be a dictionary/an object containing the 
+following items:
+
+* A client MUST set the *cookie_2* field to 16 cryptographically 
+  secure random bytes. It SHALL be different to the cookie the client 
+  currently uses, the other client currently uses and, for initiators, 
+  the *cookie_2* the responder has sent in its task's data.
+* The *exclude* field MUST contain a list/an array of WebRTC data 
+  channel ids (non-negative integers) that SHALL not be used for the 
+  signalling channel. This list MUST be available to be set from user 
+  applications that use specific data channel ids.
```

Comment:
It's for applications that use static pre-negotiated data channel ids. We don't want to get in their way.